src/utils/zcl_abapgit_progress.clas.abap

```diff
@@ -52,14 +55,31 @@ CLASS ZCL_ABAPGIT_PROGRESS IMPLEMENTATION.
 
   METHOD show.
 
-    DATA: lv_pct TYPE i.
+    DATA: lv_pct  TYPE i.
+    DATA: lv_time TYPE t.
+
+    CONSTANTS: c_wait_secs TYPE i VALUE 2.
 
     lv_pct = calc_pct( iv_current ).
 
-    CALL FUNCTION 'SAPGUI_PROGRESS_INDICATOR'
-      EXPORTING
-        percentage = lv_pct
-        text       = iv_text.
+    GET TIME.
+    lv_time = sy-uzeit.
+    IF mv_cv_time_next IS INITIAL AND mv_cv_datum_next IS INITIAL.
+      mv_cv_time_next  = lv_time.
+      mv_cv_datum_next = sy-datum.
+    ENDIF.
+
+    "We only do a progress indication if enough time has passed
+    IF lv_time  >= mv_cv_time_next  AND sy-datum = mv_cv_datum_next  OR
+       sy-datum >  mv_cv_datum_next.
+
+      CALL FUNCTION 'SAPGUI_PROGRESS_INDICATOR'
+        EXPORTING
+          percentage = lv_pct
+          text       = iv_text.
+      mv_cv_time_next = lv_time + c_wait_secs.
+
```

Comment:
also update mv_cv_datum_next?

it looks like if a user starts a process before midnight, then after midnight the indicator will show at every update instead of each 2nd second